magicctapipe/scripts/lst1_magic/coincident_brokenGPS.py

```diff
@@ -0,0 +1,418 @@
+    )
+    df_lst_subrun_file = df_lst_subrun_file.query("intensity>100")
+    df_lst_subrun_file = df_lst_subrun_file[["trigger_time", "event_type"]]
+    if i == 0:
+        df_lst_subrun_file_2 = df_lst_subrun_file
+    else:
+        df_lst_subrun_file_2 = pd.concat([df_lst_subrun_file_2, df_lst_subrun_file])
+    i = i + 1
+
+try:
+    os.mkdir(outdir)
+except FileExistsError:
+    pass
+
+# Find the time offset for each subrun combination
+for magic_subrun_file in sorted(
+    glob.glob(file_dl1_dir + "/MAGIC/dl1_MAGIC.Run" + magic_run + ".*.h5")
+):
+    data_magic = pd.read_hdf(magic_subrun_file, key="events/parameters")
+    data_magic = data_magic.query("intensity>100")
+    data_magic["trigger_time"] = (
+        data_magic["time_sec"] + data_magic["time_nanosec"] * 1e-9
+    )
+    data_magic_m1 = data_magic.query("tel_id==2")
+    data_magic_m1.reset_index(inplace=True,drop=True)
+    data_magic_m2 = data_magic.query("tel_id==3")
+    data_magic_m2.reset_index(inplace=True,drop=True)
+    magic_subrun_base = magic_subrun_file.split("/")[-1].split(".h5")[0]
+
+    data_lst = df_lst_subrun_file_2
+
+    for M1_M2 in ["M1", "M2"]:
+        if M1_M2 == "M1":
+            data_magic = data_magic_m1
+        else:
+            data_magic = data_magic_m2
+
+        data_magic.reset_index(inplace=True,drop=True)
+        min_t_magic, max_t_magic = min(data_magic["trigger_time"]), max(
+            data_magic["trigger_time"]
+        )
+        min_t_lst, max_t_lst = min(data_lst["trigger_time"]), max(
+            data_lst["trigger_time"]
+        )
+
+        if min_t_magic < min_t_lst:
+            if max_t_magic < max_t_lst:
+                data_magic = data_magic.query(
+                    "@min_t_lst < trigger_time < @max_t_magic"
+                )
+            else:
+                data_magic = data_magic.query("@min_t_lst < trigger_time < @max_t_lst")
+                data_magic.reset_index(inplace=True)#,drop=True)
+
+        if len(data_magic) > 0:
+            N_begin = data_magic.index[0] - data_magic.index[0]
+            N_final = data_magic.index[-1] - data_magic.index[0]
+
+            N_start_ = N_begin
+            N_end_ = N_start_ + 15
+
+            print(N_start_, N_end_)
+            print(data_magic[N_start_:N_end_]) 
+            # This output file will be used for the next detailed offset search
+            outfile = (
+                outdir + "/" + magic_subrun_base + "_" + str(N_start_) + "_init.npy"
+            )
+            t_magic_all, N_start_out, time_offset_best, n_coincident = find_offset(
+                data_magic, data_lst, N_start=N_start_, N_end=N_end_
+            )
+
+            if n_coincident != 0:
+                np.save(
+                    outfile.replace("MAGIC", M1_M2),
+                    np.array(
+                        [
+                            np.mean(t_magic_all),
+                            time_offset_best,
```

Comment:
why is best offset saved twice in the same array ?
 (the same also in a similar code below))